dove/src/stdout.rs

```diff
@@ -0,0 +1,213 @@
+use termcolor::{BufferWriter, Color, ColorChoice, ColorSpec, WriteColor};
+use once_cell::sync::OnceCell;
+use std::fmt::Write as FmtWriter;
+use std::io::{Write, Stdout};
+use std::sync::Mutex;
+use anyhow::Error;
+
+/// Stdout buffer for prints
+static STDOUT_STREAM: OnceCell<Mutex<Box<dyn TBufWrite + Send>>> = OnceCell::new();
+/// Marker to highlight; WHEN is 'always', 'never', or 'auto'
+static STDOUT_COLOR: OnceCell<Mutex<Box<ColorChoice>>> = OnceCell::new();
+
+/// Print output.
+pub fn print(text: &str) -> Result<(), Error> {
```

Comment:
nit: this function never returns an error, despite its signature, but panics instead